### PR TITLE
Extract user-menu component

### PR DESCRIPTION
**Pull request by Gaurav0**
Relevant to issue #127

### CODE REVIEW

**Comment by Gaurav0** [comment]
@stefanpenner @rwjblue @joostdevries Can someone review this?

File: app/templates/components/user-menu.hbs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  <span class="user-avatar"><img src="{{session.currentUser.avatarUrl32}}" /></span>
+  <li class="dropdown">
+     <a class="dropdown-toggle" data-toggle="dropdown" href="#">
+       {{session.currentUser.login}} <b class="caret"></b>
+     </a>
+     <ul class="dropdown-menu dropdown-menu-right">
+       <li><a {{action 'signOut'}} class="test-sign-out">Sign out</a></li>
+     </ul>
+  </li>
+{{else}}
```
Comment (rwjblue) [inline app/templates/components/user-menu.hbs]: Seems like the `else if session.isAuthenticated` and `else` block are roughly the same, I'd love to remove this duplication.  Looks like the following might do it:

* Add an `avatarURL` and `userName` computeds to the component that defaults to `null` if `session.isAuthenticated` is false.
* Use `{{action (if session.isAuthenticated "signOut" "signInViaGithub")}}` to get the right action.

Comment (Gaurav0) [inline app/templates/components/user-menu.hbs]: @rwjblue Duplication removed.

**Comment by rwjblue** [comment]
Thanks!

**Comment by joostdevries** [comment]
:+1:

**Pull request merged**